### ISSUE TITLE
Lock in inconsistent state on corrupted download.

### ISSUE THREAD

**Issue by Artazor** [spunjs/selenium-binaries#7]
Sometimes binaries being downloaded are corrupted. There may be many legitimate reasons why it happens, namely bad internet connection (or problems with cdn). 

Now each file is downloaded directly to its destination, and if connection breaks then corrupted partially-downloaded file sits on the place where good one should sit. And there is no escape from this state. If we run `npm install` again, it will check only presence of the files (not their validity!) and will not redownload the corrupted ones. So we need to remove them manually.

I'd propose on download give some suffix to the temporary file (like `selenium-...-smth.jar.tmp`), and only after successful download rename it to the desired destination-name.

@jsdevel, if it is ok to update versions #6 then I'd happily add this transactional logic to my pull request.

**Comment by jsdevel** [comment]
A separate PR would be best if you please.  Whatever the naming used should be fine.
